crates/bifrost/src/providers/kafka_loglet/mod.rs

```diff
@@ -324,7 +341,8 @@ impl Stream for KafkaLogletReadStream {
                             PolyBytes::Bytes(payload.payload),
                         );
                         Ok(LogEntry::new_data(
-                            KafkaLoglet::kafka_offset_to_loglet_offset(message.offset()),
+                            KafkaLoglet::kafka_offset_to_loglet_offset(message.offset())
+                                .expect("failed to get loglet offset from kafka offset"),
```

Comment:
nit: the expect message should tell why you expect it to not fail. E.g. "message.offset being small enough to fit in LogletOffset" or so.